### ISSUE TITLE
Difference in performance after loading a trained custom LSTM module (torch 1.4.0)

### ISSUE THREAD

**Issue by arshad171** [pytorch/pytorch#36433]
I have a custom LSTM module that I'm training on a dataset and then saving it, after I have attained a good accuracy on it (~ 80%).

But when I reload the model for making predictions or for continuing the training, I'm getting very different and incorrect results.
I followed the [fficial doc for saving/loading the model (https://pytorch.org/tutorials/beginner/saving_loading_models.html). None of the methods described seem to work.

Eg:
**Prediction on a sample before saving**
```
[0.3374, 0.0049, 0.2111, 0.0561, 0.3325, 0.5864, 0.1897, 0.1107, 0.0298,
0.0201, 0.0287, 0.0055, 0.0042, 0.0201, 0.0055, 0.0044]
```

**Prediction on the same sample after loading the model**
```
[0.7457, 0.0093, 0.1666, 0.0904, 0.2202, 0.4202, 0.1732, 0.1105, 0.0711,
0.0350, 0.0410, 0.0127, 0.0118, 0.0355, 0.0101, 0.0130]
```
As you can see the predictions are totally inaccurate. I had to test this a dozen times so that I could believe what I am seeing.

Here's some info on what I'm performing

### Saving the Model
After training the model, I am saving the state of the model  and the optimizer in a dict and dumping it as a .tar file

```
    state_dict = {
        'epoch': num_epochs,
        'model_state_dict': model.state_dict(),
        'optimizer_state_dict': optimizer.state_dict(),
        'loss': loss_function,
    }
   torch.save(state_dict, path)
```

### Loading the Model
 ```
   model_loaded, loss_function_loaded, optimizer_loaded = create_model()
    checkpoint = torch.load(path)
    model_loaded.load_state_dict(checkpoint['model_state_dict'])
    optimizer_loaded.load_state_dict(checkpoint['optimizer_state_dict'])

    model_loaded.eval()
   # 
   # make prediction here
   # 
```

**I have also verified the weights of the model before and after saving, the weights of all the layers seem to be correct.**

Please let me know how I can resolve the issue.

**Comment by arshad171** [comment]
There were random states involved in the model, once I used a seed or saved the state and loaded it back when evaluating, I did not see the issue.

Sorry for the invalid bug.
